R/test-app.R

```diff
@@ -86,33 +86,50 @@ testApp <- function(appDir = ".", testnames = NULL, quiet = FALSE,
 #' Prior to 1.3.1.9999, tests were stored directly in `tests/` rather than
 #' nested in `tests/shinytests/`.
 #'
+#' @param must.exist If TRUE, will error if we can't find a test directory.
+#' @param doMessage If we see that the tests are stored in the top-level tests/ directory as we used to
+#'   recommend, we will note the new recommendation in a message to the user if this is TRUE.
+#'
 #' This function does the following:
 #'  1. Check to see if `tests/shinytests/` exists. If so, use it.
 #'  2. Check to see if all the top-level R files in `tests/` appear to be shinytests. If
 #'     some are and some aren't, throw an error.
 #'  3. Assuming all top-level R files in `tests/` appear to be shinytests, return that dir.
 #' @noRd
-findTestsDir <- function(appDir) {
+findTestsDir <- function(appDir, must.exist=TRUE, doMessage=FALSE) {
```

Comment:
Can we make this `mustExist`?

Also, `doMessage` seems a bit awkward. Maybe `quiet` (defaulting to `TRUE`)? I don't feel super strongly about this though.